xen/arch/x86/domain.c

```diff
@@ -649,7 +650,8 @@ int arch_domain_create(struct domain *d, unsigned int domcr_flags,
         if ( (rc = init_domain_irq_mapping(d)) != 0 )
             goto fail;
 
-        if ( (rc = iommu_domain_init(d, false)) != 0 )
+        use_iommu = !!(domcr_flags & DOMCRF_use_iommu);
+        if ( (rc = iommu_domain_init(d, use_iommu)) != 0 )
```

Comment:
Let me repeat again what I have already said:
Basically, this and other changes (let say prediction feature) is needed for unshared IOMMUs on ARM only since the unshared IOMMUs on x86 are ok if iommu_construct() is called
later. But, in order to be more generic and for possible future optimization
make this change applicable for both platforms.
So, the my target is to make ARM happy and not to brake x86.